rest_framework/filters.py

```diff
@@ -100,12 +101,19 @@ def filter_queryset(self, request, queryset, view):
 
         orm_lookups = [self.construct_search(six.text_type(search_field))
                        for search_field in search_fields]
-
+        and_queries = []
         for search_term in self.get_search_terms(request):
             or_queries = [models.Q(**{orm_lookup: search_term})
                           for orm_lookup in orm_lookups]
-            queryset = queryset.filter(reduce(operator.or_, or_queries)).distinct()
```

Comment:
I'd prefer it if we didn't introduce the `and_queries`. We should keep each pull request as absolutely minimal as possible. Seems like there's two different changes here - the Oracle support as one, and the change in the filtering style as another. Let's just adopt the first of those two.
Apologies for the slow progress on this, but thanks for getting it nearly there! :)